pkg/pipeline/extract/aggregate/aggregate.go

```diff
@@ -167,20 +185,29 @@ func (aggregate Aggregate) Evaluate(entries []config.GenericMap) error {
 func (aggregate Aggregate) GetMetrics() []config.GenericMap {
 	var metrics []config.GenericMap
 	for _, group := range aggregate.Groups {
+		// TODO: remove prefixes when filtering is implemented in prom encode.
 		metrics = append(metrics, config.GenericMap{
-			"name":                               aggregate.Definition.Name,
-			"operation":                          aggregate.Definition.Operation,
-			"record_key":                         aggregate.Definition.RecordKey,
-			"by":                                 strings.Join(aggregate.Definition.By, ","),
-			"aggregate":                          string(group.normalizedValues),
-			"value":                              fmt.Sprintf("%f", group.value),
-			"recentRawValues":                    group.RecentRawValues,
-			"count":                              fmt.Sprintf("%d", group.count),
-			aggregate.Definition.Name + "_value": fmt.Sprintf("%f", group.value),
-			strings.Join(aggregate.Definition.By, "_"): string(group.normalizedValues),
+			"name":            aggregate.Definition.Name,
+			"operation":       aggregate.Definition.Operation,
+			"record_key":      aggregate.Definition.RecordKey,
+			"by":              strings.Join(aggregate.Definition.By, ","),
+			"aggregate":       string(group.normalizedValues),
+			"total_value":     fmt.Sprintf("%f", group.value),
+			"recentRawValues": group.RecentRawValues,
+			"total_count":     fmt.Sprintf("%d", group.count),
+			aggregate.Definition.Name + "_recent_op_value": group.recentOpValue,
+			aggregate.Definition.Name + "_recent_count":    group.recentCount,
+			aggregate.Definition.Name + "_total_value":     fmt.Sprintf("%f", group.value),
+			strings.Join(aggregate.Definition.By, "_"):     string(group.normalizedValues),
 		})
-		// Once reported, we reset the raw values accumulation
+		// Once reported, we reset the recentXXX fields
 		group.RecentRawValues = make([]float64, 0)
+		group.recentCount = 0
+		initVal, err := initValue(string(aggregate.Definition.Operation))
+		if err != nil {
+			log.Errorf("Error: %v", err)
```

Comment:
Maybe change to Panic ... this should never happen ... on the other hand, no one will see this Error ... so I suggest to panic here and also in on the other place that `initValue` is used

Reply:
OK. So in that case I can change the signature of `initValue()` to return only float64 (no error) and panic inside the function when an known operation is encountered.